devops/github-actions/docker-build-push-gcp-artifact-registry.yaml

```diff
@@ -0,0 +1,46 @@
+on:
+  push:
```

Comment:
Might be worth adding a`workflow_dispatch` option for different work styles? Then with comments like

```
on:
    workflow_dispatch:  # build new image manually
    push:  # or automatically build upon change to dependencies
        branches:
            - "main"
        paths: 
            - "Dockerfile" 
            - "requirements.txt"
```